packages/snap/src/did/jwk/jwkDidResolver.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import {
+  DIDDocument,
+  DIDResolutionOptions,
+  DIDResolutionResult,
+  DIDResolver,
+  ParsedDID,
+  Resolvable,
+  JsonWebKey,
+} from 'did-resolver';
+import { decodeBase64url } from '@veramo/utils';
+import { createTypeGuard, hasProperties } from 'create-typeguard';
+import { base64urlEncode } from '../../utils/snapUtils';
+
+const isJWK = createTypeGuard<JsonWebKey>((data) => {
```

Comment:
Maybe we can extract this into a separate shared-util library and reuse it in other parts of the monorepo.